pyvista/plotting/colors.py

```diff
@@ -348,10 +377,473 @@
+
+        >>> c = pyvista.Color("0xff000040")
+        >>> c
+        Color(hex='#ff000040')
+
+        """
+        return '#' + ''.join(
+            f"{c:0>2x}" for c in (self._red, self._green, self._blue, self._opacity)
+        )
+
+    @property
+    def name(self) -> Optional[str]:
+        """Get the color name.
+
+        Notes
+        -----
+        Refer to the table below for a list of supported colors.
+
+        .. include:: ../colors.rst
+
+        Examples
+        --------
+        Create a blue color with half opacity.
+
+        >>> import pyvista
+        >>> c = pyvista.Color("blue", default_opacity=0.5)
+        >>> c
+        Color(name='blue', hex='#0000ff80')
+
+        """
+        return self._name
+
+    @property
+    def vtk_c3ub(self) -> _vtk.vtkColor3ub:
+        """Get the color value as a VTK Color3ub instance.
+
+        Examples
+        --------
+        Create a blue color with half opacity.
+
+        >>> import pyvista
+        >>> c = pyvista.Color("blue", default_opacity=0.5)
+        >>> c
+        Color(name='blue', hex='#0000ff80')
+        >>> c.vtk_c3ub
+        vtkmodules.vtkCommonDataModel.vtkColor3ub([0, 0, 255])
+
+        """
+        return _vtk.vtkColor3ub(self._red, self._green, self._blue)
+
+    def linear_to_srgb(self):
+        """Convert from linear color values to sRGB color values.
+
+        Returns
+        -------
+        Color
+            A new Color instance with sRGB color values.
+
+        """
+        rgba = np.array(self.f_rgba)
+        mask = rgba < 0.0031308
+        rgba[mask] *= 12.92
+        rgba[~mask] = 1.055 * rgba[~mask] ** (1 / 2.4) - 0.055
+        return Color(rgba)
+
+    def srgb_to_linear(self):
+        """Convert from sRGB color values to linear color values.
+
+        Returns
+        -------
+        Color
+            A new Color instance with linear color values.
+
+        """
+        rgba = np.array(self.f_rgba)
+        mask = rgba < 0.04045
+        rgba[mask] /= 12.92
+        rgba[~mask] = ((rgba[~mask] + 0.055) / 1.055) ** 2.4
+        return Color(rgba)
+
+    @classmethod
+    def from_dict(cls, dict_):
+        """Construct from dictionary for JSON deserialization."""
+        return Color(dict_)
+
+    def to_dict(self):
+        """Convert to dictionary for JSON serialization."""
+        return (self._red, self._green, self._blue, self._opacity)
+
+    def __eq__(self, other):
+        """Equality comparison."""
+        try:
+            return self.i_rgba == Color(other).i_rgba
+        except ValueError:  # pragma: no cover
+            return NotImplemented
+
+    def __hash__(self):
+        """Hash calculation."""
+        return hash((self._red, self._green, self._blue, self._opacity))
+
+    def __repr__(self):  # pragma: no cover
+        """Human readable representation."""
+        kwargs = f"hex='{self.hex}'"
+        if self._name is not None:
+            kwargs = f"name='{self._name}', " + kwargs
```

Comment:
You might be able to replace the manual single quotes with a repr modifier, but I'm not sure it would overall be more readable:
```suggestion
        kwargs = f"hex={self.hex!r}"
        if self._name is not None:
            kwargs = f"name={self._name!r}, " + kwargs
```

Reply:
Cool!